test/PolledChannelTest.cpp

```diff
@@ -0,0 +1,159 @@
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+  #ifdef DEBUG
+  EXPECT_DEATH(packet_to_process.data(), ".*");
+  #endif
+  EXPECT_TRUE(packet_to_process.GetPacket() == nullptr);
+  EXPECT_TRUE(packet_to_process.IsEmpty());
+}
+
+TEST_F(PolledChannelTest, PolledChannelPollPacketToProcessTest) {
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", pipert::Timer::time(), 42);
+  packet_to_fill.Push();
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 1);
+  pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  EXPECT_FALSE(packet_to_process.IsEmpty());
+  EXPECT_EQ(packet_to_process.data(), 42);
+}
+
+TEST_F(PolledChannelTest, PolledChannelPollPacketToProcessWithReleaseTest) {
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", pipert::Timer::time(), 42);
+  packet_to_fill.Push();
+  pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  packet_to_process.Release();
+  EXPECT_TRUE(packet_to_process.IsEmpty());
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+}
+
+TEST_F(PolledChannelTest, PolledChannelPollPacketToProcessWithAutoReleaseTest) {
+  int channel_capacity = 2;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::PacketToFill<int> packet_to_fill = polled_channel.Acquire("TestChannel", pipert::Timer::time(), 42);
+  packet_to_fill.Push();
+  {
+    pipert::PacketToProcess<int> packet_to_process = polled_channel.Poll();
+    EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity - 1);
+    EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+  }
+  EXPECT_EQ(polled_channel.GetFreePacketSize(), channel_capacity);
+  EXPECT_EQ(polled_channel.GetQueuedPacketSize(), 0);
+}
+
+TEST_F(PolledChannelTest, PolledChannelDropPacketAfterPushTest) {
+  int channel_capacity = 1;
+  pipert::PolledChannel<int> polled_channel = scheduler_.CreatePolledChannel<int>("TestChannel", channel_capacity);
+  pipert::Timer::Time old_time = pipert::Timer::time();
+  pipert::Timer::Time new_time = pipert::Timer::time();
```

Comment:
It can theoretically happend that old and new time will be the same. In this case, the order of retrieving packets relies on the implementation. Let's hardcode this as well.